### PR TITLE
Fix URDF spawner on macOS and WSL

### LINKED ISSUE #558: URDFSpawner relative_path_prefix does not work when not using at least the parent directory of the urdf file
**Describe the Bug**
`relative_path_prefix` does not work when not using at least the parent directory of the urdf file. It must at least be the parent directory or one of it's child directory. In a real case scenario it implies that you cannot use `package://my_robot_description_package/urdf/meshes/mymesh.stl` but rather `package://urdf/meshes/mymesh.stl` for your meshes filename, messing with RViz which is not able to find the meshes.

**Steps to Reproduce**
Consider the following architectory :
```
my_robot_description_package/
└── urdf/
    ├── meshes/
    │   └── myrobot.stl
    └── myrobot.urdf
```

and the following urdf : 
```xml
<?xml version="1.0"?>
<robot name="myrobot"
  xmlns:xacro="http://ros.org/wiki/xacro">

<link name="base_footprint">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://my_robot_description_package/urdf/meshes/myrobot.stl" />
      </geometry>
    </visual>

    <collision>
      <origin xyz="0 0 0.0" rpy="0 0 0" />
      <geometry>
        <box size="0.1 0.1 0.1" />
      </geometry>
    </collision>
  </link>

</robot>
```

using the following relative path does not work :
```python
    myrobot_description_package_dir = get_package_share_directory("myrobot_description_package_dir")
    spawm_myrobot = URDFSpawner(
        name="myrobot",
        robot_description=myrobot_description
        relative_path_prefix=myrobot_description_package_dir
    )
```

This line in URDFSpawner throws Errno 2 (No such file or directory) 
```python 
shutil.copytree(relative_path_prefix, os.path.join(container_shared_folder(), os.path.basename(relative_path_prefix)))
```

Here is a workaround I found, using the parent directory of the urdf file as `relative_path_prefix` (it works with child directories too) :
```python
    myrobot_description_package_dir = get_package_share_directory("myrobot_description_package_dir")
    spawm_myrobot = URDFSpawner(
        name="myrobot",
        robot_description=myrobot_description.replace(
            "package://my_robot_description_package/", "package://"
        )
        relative_path_prefix=os.path.join(jrb_description_package_dir, "urdf")
    )
```

**Expected behavior**
Finding a way to make it work without this workaround would be great to ensure having standard urdf filenames and making RViz work.

**Affected Packages**
List of affected packages:
  - webots_ros2_driver
  

**System**
 - Webots Version: R2023a for macOS
 - ROS Version: Humble
 - Operating System: Ubuntu jammy in docker, host macOS Monterey 12.2.1, using local_simulation_server.py

### PR DESCRIPTION
**Pull request by ygoumaz**
Fixes #558 and #581.

## **#581** ##

URDFs can be spawned in 2 different ways:
- By giving the complete robot description and the path to the resources as a prefix to append to assets relative URLs.
- By giving only the path to the URDF file.

The first way was correctly handled on WSL and macOS. The second one was not and is trickier, as the path must be kept to open the file on the Linux side, while the URL should be edited to be compatible with WSL and the shared directory, respectively on Windows and macOS.

To counter this problem, on Windows and macOS, the URDF file is opened by the `ROS2_supervisor` and is passed as the robot description, and the path prefix is set according to the URDF path by searching for the 'share' directory.

The URL edition is now completely done in the supervisor, for clarity.

## **#558** ##

The URLs of xacro files in the universal robot example are reverted to the correct RVIZ format. To fix the issue on macOS, the resource folder is now copied inside another folder that has the same name as the package, to be consistent with the Rviz structure.

### CODE REVIEW

File: webots_ros2_driver/webots_ros2_driver/ros2_supervisor.py [hunk 1]
```diff
@@ -87,13 +90,77 @@ def __spawn_urdf_robot_callback(self, request, response):
...
-        # Choose the conversion according to the input
+        # Choose the conversion according to the input and platform
         if robot.urdf_path:
-            robot_string = convertUrdfFile(input=robot.urdf_path, robotName=robot_name, normal=normal,
-                                           boxCollision=box_collision, initTranslation=robot_translation,
-                                           initRotation=robot_rotation, initPos=init_pos)
+            if has_shared_folder() or is_wsl():
+                # Check that the file exists and is an URDF
+                if not os.path.isfile(robot.urdf_path):
+                    sys.exit('Input file "%s" does not exists.' % robot.urdf_path)
```
Comment (ad-daniel) [inline webots_ros2_driver/webots_ros2_driver/ros2_supervisor.py:98]:
```suggestion
                    sys.exit('Input file "%s" does not exist.' % robot.urdf_path)
```

File: webots_ros2_driver/webots_ros2_driver/ros2_supervisor.py [hunk 2]
```diff
@@ -87,13 +90,77 @@ def __spawn_urdf_robot_callback(self, request, response):
...
+                if not robot.urdf_path.endswith('.urdf'):
+                    sys.exit('"%s" is not a URDF file.' % robot.urdf_path)
```
Comment (ad-daniel) [inline webots_ros2_driver/webots_ros2_driver/ros2_supervisor.py:100]:
```suggestion
                    sys.exit('"%s" is not an URDF file.' % robot.urdf_path)
```

File: webots_ros2_driver/webots_ros2_driver/ros2_supervisor.py [hunk 3]
```diff
@@ -87,13 +90,77 @@ def __spawn_urdf_robot_callback(self, request, response):
...
+
+                # Read the content of the URDF
+                with open(robot.urdf_path, 'r') as file:
+                    urdfContent = file.read()
+                if urdfContent is None:
+                    sys.exit('Could not read the URDF file.')
```
Comment (ad-daniel) [inline webots_ros2_driver/webots_ros2_driver/ros2_supervisor.py:106]: this should probably be indented

**Review by ad-daniel — CHANGES_REQUESTED** [review]
Looks good to me, thank you.
@PymZoR may I ask you to check if the proposed patch fixes your issue?

**Comment by PymZoR** [comment]
@ad-daniel I'm going to check it this week-end

**Comment by ygoumaz** [comment]
I will merge this PR, so that we can move on with the release of version 2023.0.2.

@PymZoR feel free to mention if the fix doesn't work for you.

**Review by ad-daniel — APPROVED** [review]
Looks good to me

**Pull request merged**